Toolbox/arcsdm/pca.py

```diff
@@ -21,59 +21,61 @@
 def Execute(self, parameters, messages):
     """The source code of the tool."""
     input_data = parameters[0].valueAsText.split(';')
-    number_of_components = parameters[1].value
-    scaler_type = parameters[2].valueAsText
-    nodata_handling = parameters[3].valueAsText
-    nodata_value = parameters[4].value
+    nodata_value = parameters[1].value
+    number_of_components = parameters[2].value
+    scaler_type = parameters[3].valueAsText
+    nodata_handling = parameters[4].valueAsText
     transformed_data_output = parameters[5].valueAsText
-    principal_components_output = parameters[6].valueAsText
-    explained_variances_output = parameters[7].valueAsText
-    explained_variance_ratios_output = parameters[8].valueAsText
-
+    
     datasets = []
     for data in input_data:
+        desc = arcpy.Describe(data)
         # Load input data
-        if arcpy.Describe(data).dataType == "RasterDataset" or arcpy.Describe(data).dataType == "RasterLayer":
+        if desc.dataType == "RasterDataset" or desc.dataType == "RasterLayer":
             data = arcpy.RasterToNumPyArray(data)
-        elif arcpy.Describe(input_data).dataType == "Table":
-            data = pd.DataFrame(arcpy.da.TableToNumPyArray(data, "*"))
-        elif arcpy.Describe(data).dataType == "FeatureLayer":
+        elif desc.dataType == "Table":
+            data = pd.DataFrame(arcpy.da.TableToNumPyArray(data))
+        elif desc.dataType == "FeatureLayer":
```

Comment:
This part can now be removed, since FeatureLayer is no longer among the accepted input types.